### ISSUE TITLE
browse_surveys doesn't return include

### ISSUE THREAD

**Issue by mikesamuelson32** [tntp/surveymonkey#79]
The line at the end of the script `dplyr::select(sl, title, id, url = href, nickname)` does not allow additional columns requested through the include parameter to be passed through.  For instance, if I request 'date_modified,date_created', they are not being returned in the dataframe.  However, if the line is modified to

`dplyr::select(sl,title, id, url = href, nickname,everything())`, additional columns are returned.

**Comment by mattroumaya** [comment]
Hi @mikesamuelson32, this is a great catch! I have a solution that works as follows:

- You can provide a character vector argument for `include`, i.e., `include = c("date_modified", "date_created")`. 
- Alternatively, you can use `include = "everything"` which will return all of the non-team related parameters: `response_count, date_created, date_modified, language, question_count, analyze_url, preview`

Let me know what you think of this update, and I'll open a PR to resolve the issue tomorrow if it sounds good.

**Comment by mikesamuelson32** [comment]
Matt,

I am not seeing that either of the workarounds you list below are working.  Still getting just the 4 columns.

-- Mike

Please support my ride to end cancer.  You can make a donation at www.pelotonia.org/mikesamuelson<http://www.pelotonia.org/mikesamuelson>

________________________________

**Comment by mattroumaya** [comment]
Hi Mike, 

I haven't submitted a pull request yet - was waiting to hear back to see if this sounds like a viable solution for you. Let me pull up the project and submit a pull request and I'll let you know when you can reinstall the package and give it a shot.

**Comment by mikesamuelson32** [comment]
Matt,

Unfortunately, we have also found another issue as we have continued to dig...

The get_responses function is throwing the stop("Must specify 'oauth_token', Try using smlogin() first.") even though we are passing in a value for oauth_token.

We believe the issue is in lines 52-54, where the function calls itself recursively for each page of survey responses, but it doesn't pass in the oauth_token on those calls.

-- Mike

Please support my ride to end cancer.  You can make a donation at www.pelotonia.org/mikesamuelson<http://www.pelotonia.org/mikesamuelson>

________________________________

**Comment by mattroumaya** [comment]
Hi Mike, 

Hopefully the pull request that was just merged will fix the initial issue. 

Can you provide some more info on what you're using `get_responses()` for? To pull a survey without editing your .Rprofile with the `sm_oauth_token` global variable, you could try something like:

```
survey <- 
  123456789 %>% 
  fetch_survey_obj(oauth_token = "your very long oauth token") %>% 
  parse_survey()
```

**Comment by mikesamuelson32** [comment]
Matt,

We are actually using parse_survey, but of course it calls get_responses.  In our case, we have the oauth token stored in a database and we pull it into a variable, then we pass that into parse_survey().  The problem is that when you get into that recursive call in get_responses(), it is no longer passing in the token, so the default of getOption("sm_oauth_token"), which of course then fails for us because we are not storing the token in the .Rprofile.

This is in contrast to fetch_survey_obj, where an oauth_token passed in works fine.

-- Mike


Please support my ride to end cancer.  You can make a donation at www.pelotonia.org/mikesamuelson<http://www.pelotonia.org/mikesamuelson>

________________________________

**Comment by mattroumaya** [comment]
Are you able to assign the token to a variable within the global environment (instead of in your .Rprofile) called `sm_oauth_token`? 

If so, I think we could modify `get_responses()` so that it checks the global environment for a specific variable called `sm_oauth_token` and then assigns that to the `token` value before error-ing out. 

What do you think?

**Comment by mikesamuelson32** [comment]
We are currently passing in a variable called bearerToken in as the oauth_token parameter in parse_survey, which then passes it as oauth_token to get_responses.

In get_responses, I think you can easily fix it by changing this code, from lines 52-24

FROM THIS:
    rnext <- get_responses(id, page = page + 1, all_pages,
      per_page, start_created_at, end_created_at, start_modified_at,
      end_modified_at, sort_order, sort_by)

TO THIS:
    rnext <- get_responses(id, page = page + 1, all_pages,
      per_page, start_created_at, end_created_at, start_modified_at,
      end_modified_at, sort_order, sort_by, oauth_token = oauth_token)

-- Mike

Please support my ride to end cancer.  You can make a donation at www.pelotonia.org/mikesamuelson<http://www.pelotonia.org/mikesamuelson>

________________________________

**Comment by mattroumaya** [comment]
@mikesamuelson32 - great idea, thank you! I made that change, check it out when you can.

**Comment by mattroumaya** [comment]
@mikesamuelson32 did the PR #83 fix your issue?

**Comment by mattroumaya** [comment]
Closing this for now as it seems to be resolved from #83
